### PR TITLE
Db backup

### PR DESCRIPTION
**Pull request by kavitharaju**
- Implementation for #26 
  - Uses [ofelia](https://github.com/mcuadros/ofelia) for running cron jobs via docker containers. This helps us to keep all infrastructure in code(docker compose) base itself, not requiring any settings to be done on deployment servers.
  - The backup pattern
    - Takes a single dump within the db container every day at 00:00 am
    - Copies latest dump from DB container every day at 00:30 am
    - Retains only latest 7 of the daily backups
    - Copies latest dump on every Sunday 00:30 am
    - Retains only latest 4 of the weekly backups
    - Copies latest dump on first day of every month
    - Retains only latest 12 of the monthly dumps
    - Copies latest dump on first day of every year
    - All these backup copies are save in assistant.bible/deployment/backup. Not moving to a different server.
- Modifies the architecture.md with changes in file structure since it was created

### CODE REVIEW

**Review by woodwardmw — APPROVED** [review]
I don't know any more than you do about this, so it looks good to me...! I'm happy to merge this, and we can come back to it if there are any issues.

File: deployment/docker-compose.yml
Comment (ryderwishart) [inline deployment/docker-compose.yml:101]: @kavitharaju I am just wondering about this part:

Takes a single dump within the db container every day at 00:00 am
Copies latest dump from DB container every day at 00:30 am

Once the db get's really big (assuming that happens), is it possible that the dump might not be completed within 30 minutes? Would it make more sense to copy the latest dump just before creating a new dump (i.e., reverse these two steps). That way, a copy is made at 11:59 every day, so there are 24 hours for the copy to complete. It would amount to basically the same thing I suspect. 

This is only a suggestion, not a blocker to the PR, as we could always switch the schedule later if it caused any issues. If the dump happens very quickly then it also would not be a potential future issue as I'm envisioning.

Comment (ryderwishart) [inline deployment/docker-compose.yml:101]: Also, small type in comment: `contaier` -> `container`

File: deployment/docker-compose.yml
Comment (ryderwishart) [inline deployment/docker-compose.yml:110]: ` TZ=\` <- should there be a space here before TZ?

File: deployment/docker-compose.yml
Comment (ryderwishart) [inline deployment/docker-compose.yml:98]: I'm not familiar with all the options, but should this be `:r` ('read' I'm guessing), even though the previous volume is  `:ro` (read-only). What's the difference?

File: deployment/docker-compose.yml
Comment (ryderwishart) [inline deployment/docker-compose.yml:106]: It might be worth adding some logging in case errors show up. I asked ChatGPT what it looks like to add error logging for Ofelia jobs and it gave these examples:

For the copy jobs:
```
ofelia.job-local.copy-backup-daily.command: sh -c "cp /var/backups/postgres_backup_latest.sql /var/backup-copies/postgres_backup_daily_`TZ=\":Asia/Calcutta\" date +%Y-%m-%d_%H_%M_%S`.sql && echo 'Copy successful' >> /var/log/backup.log || echo 'Copy failed' >> /var/log/backup.log"
```

For the prune jobs:
```
ofelia.job-local.prune-daily-backup.command: sh -c "ls -tr /var/backup-copies/postgres_backup_daily* | head -n -7 | xargs rm -f -- && echo 'Prune successful' >> /var/log/backup.log || echo 'Prune failed' >> /var/log/backup.log"
```

We might need/want a specific command for the log files.

This is just an idea. I'll defer to your judgement and expertise!

File: deployment/docker-compose.yml
Comment (ryderwishart) [reply deployment/docker-compose.yml:110]: Down on line 114 as well. 

I ask because line 102 does not have this space.

**Review by ryderwishart — APPROVED** [review]
A few queries/suggestions, which may not be applicable. Otherwise LGTM. Great work!

File: deployment/docker-compose.yml
Comment (kavitharaju) [reply deployment/docker-compose.yml:101]: If new dump gets created during copy is going on, would that lead to an issue?

File: deployment/docker-compose.yml
Comment (kavitharaju) [reply deployment/docker-compose.yml:110]: That space wasn't an issue. But removed them anyway.

File: deployment/docker-compose.yml
Comment (kavitharaju) [reply deployment/docker-compose.yml:98]: There is no `r`, it should be `ro`. :+1:

**Comment by kavitharaju** [comment]
Thanks @ryderwishart  for looking into it. Made following updates
- `r` to `ro` in volumes
- Typo fix for "container"
- Schedule is now. 
  - DB dump at 00:00 am
  - Pruning at 2:45 am
  - Backup copying at 3:00 am
- Logging for backup and pruning tasks

File: deployment/docker-compose.yml
Comment (kavitharaju) [reply deployment/docker-compose.yml]: ChatGPT's suggestions were very helpful :smile:

**Pull request merged**
